brainiak/reprsimil/brsa.py

```diff
@@ -274,9 +275,11 @@ def __init__(
                                           max_iter=20, tol=tol)
         elif nureg_method == 'ICA':
             self.nureg_method = FastICA(n_components=n_nureg, whiten=True)
-        else:
+        elif type(nureg_method) is str:
             raise ValueError('nureg_method can only be FA, PCA, '
                              'SPCA(for sparse PCA) or ICA')
+        else:
+            self.nureg_method = nureg_method
```

Comment:
I think accepting strings and instantiated objects is worse than only accepting strings, because it makes reasoning about the code more complicated. Please revert to the initial code that only accepts strings.